Jint/Engine.cs

```diff
@@ -717,13 +722,7 @@ public void PutValue(Reference reference, JsValue value)
             else
             {
                 var baseValue = reference.GetBase();
-                if (!(baseValue is EnvironmentRecord record))
```

Comment:
I assume that if there is this test it's because it was in the spec. Maybe there is not code path for it, but I wouldn't remove it, unless you found out it could never be true.

Reply:
But as you can see the code right after it does forced cast so the invariant is present. It just changes from ambiguous ArgumentNullException to maybe even better InvalidCastException?

Reply:
Sorry, I thought it was throwing a JS exception. It's just dotnet one.